astropy/coordinates/baseframe.py

```diff
@@ -265,28 +292,36 @@ class BaseCoordinateFrame(ShapedLikeNDArray, metaclass=FrameMeta):
     frame_attributes = OrderedDict()
     # Default empty frame_attributes dict
 
-    def __init__(self, *args, copy=True, representation=None,
-                 differential_cls=None,**kwargs):
+    def __init__(self, *args, copy=True, representation_type=None,
+                 differential_type=None, **kwargs):
         self._attr_names_with_defaults = []
 
-        # TODO: we should be able to deal with an instance, not just a
-        # class or string for representation and differential_cls.
+        # This is here for backwards compatibility. It should be possible
+        # to use either the kwarg representation_type, or representation.
+        # TODO: In future versions, we will raise a deprecation warning here:
+        _normalize_representation_type(kwargs)
```

Comment:
this looks to me to be the source of all the failures: `_normalize_representation_type` is expecting `representation_type` to be in the `kwargs`, but it isn't - it's a separate stand-alone keyword.  If I add this immediately above:
```
        if 'representation' in kwargs:
            representation_type = kwargs.pop('representation')
```
most of the tests begin to pass.  So from that I derived the commit I pushed up to this branch that I hope should get the tests all passing.